wordpress-mobile/WordPress-Android
# [Jetpack Social] Post settings endpoint integration

**Should be merged after https://github.com/wordpress-mobile/WordPress-Android/pull/18781**

This is the first part of the post settings integration with `/jetpack-social` endpoint.

No connections 
<img width="324" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/02b8dabf-494a-4dda-b5e1-c4664d7389b9">

With connections and share limit enabled
<img width="308" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/539cf480-5fce-47ff-a635-edd10361b99f">

With connections and share limit disabled should only display the list of connections and the message container.

Part of #18571 

To test JP social in post settings:
1 - Enable `JetpackSocialFeatureConfig` feature flag;
2 - Select a site that is not self-hosted and has at least one social sharing connection;
3 - Open post settings: jetpack social section should be shown, the sharing connection listed and message container is shown. Make sure the displayed UI elements are correct. This is a WIP, so the buttons are not working yet;
4 - Select a site that is self-hosted and has jp social plugin (free version) installed;
5 - Open post settings: jetpack social section should be shown , sharing connection listed **and** share limit layout displayed. Make sure the displayed UI elements are correct. Some labels of the share limit layout are still placeholders and will be replaced in future PRs;
6 - Select a site that has no social sharing connections;
7 - Open post settings: jetpack social section should be shown with the no connections layout. Verify that the the UI elements are correct;
8 - Disable `JetpackSocialFeatureConfig` feature flag;
9 - Open post settings again: jetpack social section should **not** be shown.

To test `PublicizeTable` migration:
1 - Install an older version of WP/JP;
2 - Open "My site" -> "MENU" -> "Sharing" screen (so `PublicizeTable` is created);
3 - Install the version from this branch;
4 - Enable `JetpackSocialFeatureConfig` feature flag;
5 - Open "My site" -> "Posts" -> Select a post -> Post settings screen and make sure it doesn't crash;

6 - Clear app data;

7 - Install an older version of WP/JP;
8 - Do not open "Sharing" screen like step 2
9 - Install the version from this branch;
10 - Enable `JetpackSocialFeatureConfig` feature flag;
11 - Open "My site" -> "Posts" -> Select a post -> Post settings screen and make sure it doesn't crash;

12 - Clear app data;

13 - Uninstall the app;
14 - Install the version from this branch;
15 - Enable `JetpackSocialFeatureConfig` feature flag;
16 - Open "My site" -> "Posts" -> Select a post -> Post settings screen and make sure it doesn't crash;


## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit testing

3. What automated tests I added (or what prevented me from doing so)
`EditPostPublishSettingsJetpackSocialUiStateMapperTest.kt`, `EditPostPublishSettingsViewModelTest.kt`, `PublishSettingsViewModelTest.kt`, `GetPublicizeConnectionsForUserUseCaseTest.kt`

PR submission checklist:


UI Changes testing checklist:
